File: 1.0/src/cz/mg/vulkan/PFNvkDestroyImage.java
package cz.mg.vulkan;

public class PFNvkDestroyImage extends VkFunctionPointer {
    public PFNvkDestroyImage() {
    }

    public PFNvkDestroyImage(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkDestroyImage(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDestroyImage(long value) {
        setValue(value);
    }

    public PFNvkDestroyImage(VkInstance instance) {
        super(instance, new VkString("vkDestroyImage"));
    }

    public void call(VkDevice device, VkImage image, VkAllocationCallbacks pAllocator){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddress(), image != null ? image.getVkAddress() : VkPointer.getNullAddress(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    protected static native void call(long vkaddress, long device, long image, long pAllocator);
}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkDestroyImage.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkDestroyImage_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong image, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkDestroyImage f = (PFN_vkDestroyImage)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkImage*)jniLongToPointer(image)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}

